src/modes/direct_select.js

```diff
@@ -107,6 +109,22 @@ module.exports = function(ctx, opts) {
         geojson.properties.active = 'false';
         push(geojson);
       }
+    },
+    trash: function(options = {}) {
+      if (selectedCoordPaths.length === 0) {
+        return ctx.events.changeMode('simple_select', { features: [feature] }, { emit: options.emit });
+      }
+
+      selectedCoordPaths.sort().reverse().forEach(id => feature.removeCoordinate(id));
+      ctx.map.fire(Constants.events.UPDATE, {
+        type: Constants.updateTypes.CHANGE_COORDINATES,
+        features: ctx.store.getSelected().map(f => f.toGeoJSON())
+      });
+      selectedCoordPaths = [];
+      if (feature.isValid() === false) {
+        ctx.store.delete([featureId], { emit: options.emit });
```

Comment:
Any `emit` left in the code is a relic of earlier attempts and should be destroyed. Only `silent` should remain.